core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -261,14 +261,48 @@ public static void createTable(
             int tableVersion,
             int tableId
     ) {
-        LOG.debug().$("create table [name=").$(tableDir).$(']').$();
-        path.of(root).concat(tableDir);
+        createTable(ff, root, mkDirMode, memory, path, false, tableDir, structure, tableVersion, tableId);
+    }
 
-        if (ff.mkdirs(path.slash$(), mkDirMode) != 0) {
-            throw CairoException.critical(ff.errno()).put("could not create [dir=").put(path).put(']');
+    public static void createTable(
+            FilesFacade ff,
+            CharSequence root,
+            int mkDirMode,
+            MemoryMARW memory,
+            Path path,
+            boolean pathIsLoadedWithVolume,
+            CharSequence tableDir,
+            TableStructure structure,
+            int tableVersion,
+            int tableId
+    ) {
+        Path normalPath = null;
+        if (!pathIsLoadedWithVolume) {
+            path.of(root).concat(tableDir).$();
+            LOG.info().$("create table [name=").utf8(tableDir).I$();
+        } else {
+            // path has been set by CREATE TABLE ... [IN VOLUME 'path'].
+            // it is a valid folder, or link to a folder, checked at bootstrap
+            normalPath = Path.getThreadLocal2(root).concat(tableDir).$();
+            LOG.info().$("create table in volume [path=").utf8(normalPath).I$();
+            if (ff.isDirOrSoftLinkDir(normalPath)) {
+                throw CairoException.critical(ff.errno()).put("table folder already exists in volume [path=").put(normalPath).put(']');
```

Comment:
nit: in other places in our code we use `directory`, not `folder`. Folder is a Windows term while we tend to be more POSIX friendly.

Reply:
apologies, the Windows environment tripped me, I coded this on the standing desk, :), will rename.